### PR TITLE
#164046256 Users Should Follow Each Other

### PR DESCRIPTION
**Pull request by BrianSerem**
**Description**

This Pull request implements users' abiity to follow each other, an existing user may be followed by another user. 
These endpoints will now be working:

**GET** `/api/profiles/<username>/follow/`                        - Follow a user
**DELETE** `/api/profiles/<username>/follow/`      - Unfollow a user
**GET** `/api/profiles/<username>/following/`       - Get a users statistics: followers/followees
<hr>
**Type of change**

- [x] new feature(non-breaking change which adds functionality).

<hr>

**How has this been tested**

- [x] End to End
- [x] Integration
<hr>

**How to test this manually**
Using Postman, navigate to [https://ah-legion-staging-pr-22.herokuapp.com](https://ah-legion-staging-pr-18.herokuapp.com), use instructions below to test the various endpoints.

##### : Follow a user
[https://ah-legion-staging-pr-22.herokuapp.com/api/profiles/](https://ah-legion-staging-pr-18.herokuapp.com/api/profiles/username/follow/)
##### Unfollow a user: 
[https://ah-legion-staging-pr-22.herokuapp.com/api/profiles/username/](https://ah-legion-staging-pr-18.herokuapp.com/api/profiles/username/)
##### Get user statistics: Number of followers and followees:
 [https://ah-legion-staging-pr-22.herokuapp.com/api/profiles/username/followings/](https://ah-legion-staging-pr-18.herokuapp.com/api/profiles/username/followings/)

<hr>
**Checklist:**

- [x] This follows all the guidelines for this project.

**Related Stories**
<hr>

[#164046256](https://www.pivotaltracker.com/n/projects/2313280/stories/164046256)

### CODE REVIEW

**Review by Etomovich — APPROVED** [review]

**Review by RachelleMaina — APPROVED** [review]

**Review by cob04 — APPROVED** [review]
looks good

**Review by MandelaK — APPROVED** [review]

**Review by ElMonstro — APPROVED** [review]
Approved

File: authors/apps/authentication/tests/test_social_auth.py [hunk 1]
```diff
@@ -59,8 +61,8 @@ def test_facebook_oauth(self):
     def test_google_auth(self):
         """Test Google Oauth"""
         self.data = {
-            "access_token": os.environ['GOOGLE_ACCESS_TOKEN'],
-            "access_token_secret": os.environ['TWITTER_ACCESS_TOKEN_SECRET'],
```
Comment (abulojoshua1) [inline authors/apps/authentication/tests/test_social_auth.py]: @BrianSerem, please remove the fixes that do not contribute to the PT board story. It is better to have this in a separate PR.

File: authors/settings/base.py [hunk 2]
```diff
@@ -149,16 +149,16 @@
...
+SOCIAL_AUTH_TWITTER_KEY = config('SOCIAL_AUTH_TWITTER_KEY')
+SOCIAL_AUTH_TWITTER_SECRET = config('SOCIAL_AUTH_TWITTER_SECRET')
 SOCIAL_AUTH_TWITTER_SCOPE = ['email']
 
-SOCIAL_AUTH_FACEBOOK_KEY = os.environ['SOCIAL_AUTH_FACEBOOK_KEY']
-SOCIAL_AUTH_FACEBOOK_SECRET = os.environ['SOCIAL_AUTH_FACEBOOK_SECRET']
+SOCIAL_AUTH_FACEBOOK_KEY = config('SOCIAL_AUTH_FACEBOOK_KEY')
+SOCIAL_AUTH_FACEBOOK_SECRET = config('SOCIAL_AUTH_FACEBOOK_SECRET')
 SOCIAL_AUTH_FACEBOOK_SCOPE = ['email']
 SOCIAL_AUTH_FACEBOOK_PROFILE_EXTRA_PARAMS = {
```
Comment (abulojoshua1) [inline authors/settings/base.py]: @BrianSerem, same thing here.

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/tests/test_social_auth.py (hunk 1 above)
Comment (BrianSerem) [reply authors/apps/authentication/tests/test_social_auth.py]: Okay, I have removed the unrelated changes.

File: authors/settings/base.py (hunk 2 above)
Comment (BrianSerem) [reply authors/settings/base.py]: I have implemented the changes requested, let me know if you need any other changes made.

**Review by abulojoshua1 — APPROVED** [review]

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]
@BrianSerem, Please fix the failing build.

**Comment by BrianSerem** [comment]
> @BrianSerem, Please fix the failing build.

Was waiting for @ElMonstro 's PR to be merged, it fixed the failing build

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**